units/src/amount/unsigned.rs

```diff
@@ -227,12 +247,19 @@ impl Amount {
     ///
     /// Returns [`None`] if overflow occurred.
     #[cfg(feature = "alloc")]
-    pub fn checked_div_by_weight_ceil(self, rhs: Weight) -> Option<FeeRate> {
-        let sats = self.0.checked_mul(1_000)?;
-        let wu = rhs.to_wu();
-
-        let fee_rate = sats.checked_add(wu.checked_sub(1)?)?.checked_div(wu)?;
-        Some(FeeRate::from_sat_per_kwu(fee_rate))
+    pub const fn checked_div_by_weight_ceil(self, rhs: Weight) -> Option<FeeRate> {
+        // No `?` operator in const context.
+        if let Some(sats) = self.0.checked_mul(1_000) {
+            let wu = rhs.to_wu();
```

Comment:
In 01b93398e28d2f73323bf11859e7a2fdf7005da2:

Can you please move this line to the top of the function. It has tripped me up multiple times now (here and previously in https://github.com/rust-bitcoin/rust-bitcoin/pull/3430#issuecomment-2398593256 ... and I even remembered the previous instance).

The problem is that you are defining `sats` and then immediately doing a `to_wu` conversion. To a quick/lazy reader it looks like you are converting the just-computed `sats` rather than the unrelated `rhs`.

Reply:
I'm not bothered by this really since `wu` is being defined right before it's use.  Moving the assignment to the beginning of the function seems fine too since it's used throughout.

Reply:
Done.

I did think that too, but thought probably incorrectly if the first `if let` fails then that line never gets run so is slightly more efficient.

Reply:
I was also going to remark too that the compiler dictates the efficiency here most likely.